fdbclient/BlobCipher.cpp

```diff
@@ -369,30 +430,52 @@ BlobCipherKey::BlobCipherKey(const EncryptCipherDomainId& domainId,
 	// timestamp.
 	ASSERT(refreshAt == std::numeric_limits<int64_t>::max() || (refreshAt <= expireAt));
 
-	initKey(domainId, baseCiph, baseCiphLen, baseCiphId, salt, refreshAt, expireAt);
+	initKey(domainId, baseCiphId, baseCiph, baseCiphLen, baseCiphKCV, salt, refreshAt, expireAt);
 }
 
 BlobCipherKey::BlobCipherKey(const EncryptCipherDomainId& domainId,
                              const EncryptCipherBaseKeyId& baseCiphId,
                              const uint8_t* baseCiph,
                              const int baseCiphLen,
+                             const EncryptCipherKeyCheckValue baseCiphKCV,
                              const EncryptCipherRandomSalt& salt,
                              const int64_t refreshAt,
                              const int64_t expireAt) {
-	initKey(domainId, baseCiph, baseCiphLen, baseCiphId, salt, refreshAt, expireAt);
+	initKey(domainId, baseCiphId, baseCiph, baseCiphLen, baseCiphKCV, salt, refreshAt, expireAt);
 }
 
 void BlobCipherKey::initKey(const EncryptCipherDomainId& domainId,
+                            const EncryptCipherBaseKeyId& baseCiphId,
                             const uint8_t* baseCiph,
                             const int baseCiphLen,
-                            const EncryptCipherBaseKeyId& baseCiphId,
+                            const EncryptCipherKeyCheckValue baseCiphKCV,
                             const EncryptCipherRandomSalt& salt,
                             const int64_t refreshAt,
                             const int64_t expireAt) {
+	if (baseCiphLen > MAX_BASE_CIPHER_LEN) {
+		// HMAC_SHA digest generation accepts upto MAX_BASE_CIPHER_LEN key-buffer, longer keys are truncated and weakens
+		// the security guarantees.
+		TraceEvent(SevWarnAlways, "MaxBaseCipherKeyLimit")
```

Comment:
Providing more than longer key is a serious error as it undermines the security guarantees by the algorithm. All tests are modified to never supply a key of this sounds, making it SevError exposes such instances if added due to regression.

Reply:
My point is given `baseCipher` is an external input to the system, it should not be SevError, anyone injecting unacceptable cipherLen would observe the operation as failed, but, it is not a system error. 

Also, not sure what is the gain we get from `SevError`?

Reply:
It's easier to identify this in production environments. We track Sev40s much more seriously than Sev30s so it would be a much clearer indication of what happened instead of some operation failing and having to deduce what went wrong

Reply:
We have added dedicated error codes for the same reason, for instance: both BlobCipher and RESTKmsConnector would throw an dedicated error-code to the callers to make triaging easy.